### PR TITLE
Use unstable_batchedUpdates to group setState

### PR DESCRIPTION
**Pull request by kwohlfahrt**
Previously, react did some work immediately after setState was called.
If a single action triggered state changes in many elements, this caused
updates to be quite slow - up to 1000x vs Redux.

unstable_batchedUpdates is used by Redux to avoid this:
https://react-redux.js.org/api/batch#references.

### CODE REVIEW

**Comment by kwohlfahrt** [comment]
@tdawes: I don't know how to actually test the performance of react rendering in an automated way. I've included the benchmark example for now so we can at least do some manual testing.

**Comment by tdawes** [comment]
We might want to move the benchmark examples into its own directory. _examples/_ should really be documentation (we should probably get rid of a few of the less useful examples).

File: examples/benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Prodo Clock
```
[the remark this answers is not in the document]
Comment (kwohlfahrt) [reply examples/benchmark/README.md]: Done.

**Comment by kwohlfahrt** [comment]
Agree with @tdawes in principle, but not in scope here. I'll merge when tests pass.

**Pull request merged**